modulecheck-plugin/src/test/kotlin/modulecheck/gradle/AndroidBuildConfigAvoidanceTest.kt

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (C) 2021-2022 Rick Busarow
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package modulecheck.gradle
+
+import modulecheck.specs.DEFAULT_AGP_VERSION
+import modulecheck.specs.DEFAULT_KOTLIN_VERSION
+import modulecheck.testing.createSafely
+import modulecheck.utils.child
+import org.junit.jupiter.api.Test
+
+class AndroidBuildConfigAvoidanceTest : BasePluginTest() {
+
+  @Test
+  fun `buildConfig task should be required when not configured in android library module`() {
+    androidProject(":", "com.modulecheck.lib1") {
+      buildFile {
+        """
+        plugins {
+          id("com.android.library") version "$DEFAULT_AGP_VERSION"
+          kotlin("android") version "$DEFAULT_KOTLIN_VERSION"
+          id("com.rickbusarow.module-check")
+        }
+        repositories {
+          mavenCentral()
+          google()
+          jcenter()
+          maven("https://plugins.gradle.org/m2/")
+          maven("https://oss.sonatype.org/content/repositories/snapshots")
+        }
+
+        android {
+          defaultConfig {
+            minSdkVersion(23)
+            compileSdkVersion(30)
+            targetSdkVersion(30)
+          }
+        }
+        """
+      }
+
+      projectDir.child("settings.gradle.kts").createSafely()
+    }
+
+    shouldSucceed("moduleCheck").apply {
+      // Assert that nothing else executed.
+      // If ModuleCheck were relying upon buildConfig tasks, they'd be in this list.
+      tasks.map { it.path } shouldBe listOf(
+        ":preBuild",
+        ":preDebugAndroidTestBuild",
+        ":preDebugBuild",
+        ":extractDeepLinksDebug",
+        ":processDebugManifest",
+        ":generateDebugBuildConfig",
+        ":preReleaseBuild",
+        ":generateReleaseBuildConfig",
+        ":processDebugAndroidTestManifest",
+        ":generateDebugAndroidTestBuildConfig",
+        ":moduleCheck"
+      )
+    }
+  }
+
+  @Test
+  fun `buildConfig task should not be required when disabled in android library module`() {
```

Comment:
What if it is disabled via gradle.properties. Does this handle that automatically?

Reply:
`gradle.properties`, `-Pandroid.defaults.buildfeatures.buildconfig=false`, and the dsl are all functionally the same.  The dsl extensions all just delegate to `project.properties`.

Reply:
Didn't know. That's very neat